unstructured_ingest/v2/processes/connectors/mongodb.py

```diff
@@ -53,6 +62,166 @@ class MongoDBUploadStagerConfig(UploadStagerConfig):
+                        "document_id": str(doc_id),
+                    },
+                )
+
+                file_data = FileData(
+                    identifier=str(doc_id),
+                    connector_type=self.connector_type,
+                    source_identifiers=source_identifiers,
+                    metadata=metadata,
+                    additional_metadata={},
+                )
+                yield file_data
+
+
+@dataclass
+class MongoDBDownloader(Downloader):
+    download_config: MongoDBDownloaderConfig
+    connection_config: MongoDBConnectionConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def create_client(self) -> "MongoClient":
+        from pymongo import MongoClient
+        from pymongo.driver_info import DriverInfo
+        from pymongo.server_api import ServerApi
+
+        access_config = self.connection_config.access_config.get_secret_value()
+
+        if access_config.uri:
+            return MongoClient(
+                access_config.uri,
+                server_api=ServerApi(version=SERVER_API_VERSION),
+                driver=DriverInfo(name="unstructured", version=unstructured_version),
+            )
+        else:
+            return MongoClient(
+                host=self.connection_config.host,
+                port=self.connection_config.port,
+                server_api=ServerApi(version=SERVER_API_VERSION),
+            )
+
+    @SourceConnectionError.wrap
+    @requires_dependencies(["pymongo"], extras="mongodb")
+    def run(self, file_data: FileData, **kwargs: Any) -> download_responses:
+        """Fetches the document from MongoDB and writes it to a file."""
+        from bson.objectid import ObjectId
+
+        client = self.create_client()
+        database = client[self.connection_config.database]
+        collection = database[self.connection_config.collection]
+
+        # Convert the identifier back to an ObjectId if necessary
+        try:
+            document_id = ObjectId(file_data.identifier)
+        except Exception:
+            document_id = file_data.identifier  # Handle non-ObjectId identifiers
+
+        try:
+            doc = collection.find_one({"_id": document_id})
+        except Exception as e:
+            logger.error(f"Failed to fetch document with ID {document_id}: {e}", exc_info=True)
+            raise e
+
+        if doc is None:
+            raise FileNotFoundError(f"Document with ID {document_id} not found")
+
+        doc.pop("_id", None)
+
+        flattened_dict = flatten_dict(dictionary=doc)
+        concatenated_values = "\n".join(str(value) for value in flattened_dict.values())
+
+        download_path = self.get_download_path(file_data)
+        if download_path is None:
+            raise ValueError("Download path could not be determined")
+
+        download_path.parent.mkdir(parents=True, exist_ok=True)
+
+        with open(download_path, "w", encoding="utf8") as f:
```

Comment:
Let's add `.txt` to the files it creates in the download step. Right now its writing files with no extension. @rbiseck3 would you agree?

Reply:
Done, added .txt in downloader plus modified test data to match metadata.filename accordingly